test/functional/test-error.js

```diff
@@ -131,46 +139,62 @@ describe('maybeReportErrorToViewer', () => {
     sendMessageStub = sandbox.stub(viewer, 'sendMessage');
 
     sandbox.stub(Services, 'viewerForDoc').returns(viewer);
+
+    xhrRequests = [];
+    setupMockXhr();
   });
 
   afterEach(() => {
     sandbox.restore();
   });
 
-  it('should not report if AMP doc is not single', () => {
+  it('should report to server if AMP doc is not single', () => {
     ampdocServiceForStub.returns({isSingleDoc: () => false});
-    return maybeReportErrorToViewer(win, data)
-        .then(() => expect(sendMessageStub).to.not.have.been.called);
+    return reportErrorToServerOrViewer(win, data)
+        .then(() => {
+          expect(xhrRequests).to.have.lengthOf(1);
+          expect(sendMessageStub).to.not.have.been.called;
+        });
   });
 
-  it('should not report if AMP doc is not opted in', () => {
+  it('should report to server if AMP doc is not opted in', () => {
     const nonOptedInDoc =
-          window.document.implementation.createHTMLDocument('');
+      window.document.implementation.createHTMLDocument('');
     ampdocServiceForStub.returns({
       isSingleDoc: () => true,
       getAmpDoc: () => ({getRootNode: () => nonOptedInDoc}),
     });
-    return maybeReportErrorToViewer(win, data)
-        .then(() => expect(sendMessageStub).to.not.have.been.called);
+    return reportErrorToServerOrViewer(win, data)
+        .then(() => {
+          expect(xhrRequests).to.have.lengthOf(1);
+          expect(sendMessageStub).to.not.have.been.called;}
```

Comment:
Formatting looks a bit weird here - `}` should be on the next line.

Reply:
fixed.